### PR TITLE
Fix crosswalk warning too early

### PR DESCRIPTION
**Pull request by Dima-1**

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java [hunk 1]
```diff
@@ -386,6 +384,9 @@ public void announceVisibleLocations() {
 					while (kIterator < lp.size() && lp.get(kIterator).routeIndex < currentRoute) {
 						if (type == ALARMS) {
 							AlarmInfo alarm = (AlarmInfo) lp.get(kIterator).getPoint();
+							// TODO test tunnels that alarm in the tunnel works correctly
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java:387]: To do not ok in result code

File: OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java [hunk 2]
```diff
@@ -244,29 +244,27 @@ public AlarmInfo getMostImportantAlarm(SpeedConstants sc, boolean showCameras) {
...
-					if (inf.getLocationIndex() == currentRoute && d > 10) {
-						return null;
+					if (inf.getLocationIndex() < currentRoute) {
+						// update remaining length
+						inf.setFloatValue(route.getDistanceToPoint(lastKnownLocation, inf.getLastLocationIndex()));
+					} else {
+						distanceByRoute = route.getDistanceToPoint(lastKnownLocation, inf.getLocationIndex() - 1);
 					}
-					if (!atd.isTurnStateActive(0, d, STATE_LONG_PNT_APPROACH)) {
+					if (!atd.isTurnStateActive(0, distanceByRoute, STATE_SHORT_PNT_APPROACH)) { // TODO or STATE_LONG_PNT_APPROACH
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java:259]: Todo not ok in result code

Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java]: If to change STATE_LONG_PNT_APPROACH to the STATE_SHORT_PNT_APPROACH, will appear issue  https://github.com/osmandapp/OsmAnd/issues/3772

File: OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java (hunk 1 above)
Comment (Dima-1) [reply OsmAnd/src/net/osmand/plus/helpers/WaypointHelper.java]: Fix tunnels announces

**Pull request merged**